implementations/rust/ockam/ockam_api/src/authenticator/storage/authority_enrollment_token_repository_sql.rs

```diff
@@ -0,0 +1,234 @@
+use ockam::identity::TimestampInSeconds;
+use sqlx::*;
+use tracing::debug;
+
+use ockam_core::async_trait;
+use ockam_core::Result;
+use ockam_node::database::{FromSqlxError, SqlxDatabase, ToSqlxType, ToVoid};
+
+use crate::authenticator::one_time_code::OneTimeCode;
+use crate::authenticator::{AuthorityEnrollmentTokenRepository, Token, TokenRow};
+
+/// Implementation of [`AuthorityEnrollmentTokenRepository`] trait based on an underlying database
+/// using sqlx as its API, and Sqlite as its driver
+#[derive(Clone)]
+pub struct AuthorityEnrollmentTokenSqlxDatabase {
+    database: SqlxDatabase,
+}
+
+impl AuthorityEnrollmentTokenSqlxDatabase {
+    /// Create a new database
+    pub fn new(database: SqlxDatabase) -> Self {
+        debug!("create a repository for authority enrollment tokens");
+        Self { database }
+    }
+
+    /// Create a new in-memory database
+    pub async fn create() -> Result<Self> {
+        Ok(Self::new(
+            SqlxDatabase::in_memory("authority enrollment tokens").await?,
+        ))
+    }
+}
+
+#[async_trait]
+impl AuthorityEnrollmentTokenRepository for AuthorityEnrollmentTokenSqlxDatabase {
+    async fn use_token(
+        &self,
+        one_time_code: OneTimeCode,
+        now: TimestampInSeconds,
+    ) -> Result<Option<Token>> {
+        let query1 =
```

Comment:
It would be good to add a comment to explain why we are doing this (purge the database of expired tokens)